mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java

```diff
@@ -191,16 +191,16 @@ private ACLDiff updateAcl(CassandraId cassandraId, Function<ACLWithVersion, ACLW
                 conditionalInsertStatement.bind()
                     .setUUID(CassandraACLTable.ID, cassandraId.asUuid())
                     .setString(CassandraACLTable.ACL, MailboxACLJsonConverter.toJson(acl)))
-                .thenApply(Optional::of)
-                .thenApply(optional -> optional.filter(b -> b).map(any -> acl))
-                .join();
+                .map(Optional::of)
+                .map(optional -> optional.filter(b -> b).map(any -> acl))
+                .block();
         } catch (JsonProcessingException exception) {
             throw new RuntimeException(exception);
         }
     }
 
     private Optional<ACLWithVersion> getAclWithVersion(CassandraId cassandraId) {
-        ResultSet resultSet = getStoredACLRow(cassandraId).join();
+        ResultSet resultSet = getStoredACLRow(cassandraId).block();
```

Comment:
Does this means that "`we block every time we read an ACL`" ?

Can't it be the source of slooooow mailboxes reads?

Can we try to get read of that 'block' here?

This should be fairly easy...